### PR TITLE
No group filter in devices

### PR DESCRIPTION
**Pull request by ct-goranpalibrk**
This PR adds new option "no group" into Device filter panel. Issue #1218

### CODE REVIEW

**Comment by LeoNerdoG** [comment]
I have verified that this PR successfully resolves issue #1218.

File: console/module/device/src/main/java/org/eclipse/kapua/app/console/module/device/client/device/DeviceFilterPanel.java [hunk 1]
```diff
@@ -66,7 +66,7 @@
     private final KapuaTextField<String> customAttribute2Field;
 
     private ComboBox<GwtGroup> groupsCombo;
-    private GwtGroup allGroup;
+    private GwtGroup allGroup, noGroup;
```
Comment (muros-ct) [inline console/module/device/src/main/java/org/eclipse/kapua/app/console/module/device/client/device/DeviceFilterPanel.java]: Slit into two lines.

File: console/module/device/src/main/java/org/eclipse/kapua/app/console/module/device/shared/model/GwtDeviceQueryPredicates.java [hunk 2]
```diff
@@ -55,6 +55,13 @@
         }
     }
 
+    public enum GwtGroupDevice implements IsSerializable {
+        NO_GROUP, ANY;
+
+        private GwtGroupDevice() {
```
Comment (muros-ct) [inline console/module/device/src/main/java/org/eclipse/kapua/app/console/module/device/shared/model/GwtDeviceQueryPredicates.java]: Only this enum constructor is private. Why?

File: console/module/device/src/main/java/org/eclipse/kapua/app/console/module/device/shared/model/GwtDeviceQueryPredicates.java [hunk 3]
```diff
@@ -218,6 +227,18 @@ public GwtDeviceCertificateStatus getDeviceCertificateStatusEnum() {
         return get("deviceCertificateStatusEnum");
     }
 
+    public String getGroupDevice() {
+        return get("groupDevice");
+    }
+
+    public GwtGroupDevice getGroupDeviceStatusEnum() {
```
Comment (muros-ct) [inline console/module/device/src/main/java/org/eclipse/kapua/app/console/module/device/shared/model/GwtDeviceQueryPredicates.java]: Why StatusEnum is this copy paste issue?

File: console/module/device/src/main/java/org/eclipse/kapua/app/console/module/device/shared/util/GwtKapuaDeviceModelConverter.java [hunk 4]
```diff
@@ -187,8 +187,16 @@ public static DeviceQuery convertDeviceQuery(PagingLoadConfig loadConfig, GwtDev
                 andPred = andPred.and(new AttributePredicate<DeviceConnectionStatus>(DevicePredicates.CONNECTION_STATUS, DeviceConnectionStatus.valueOf(predicates.getDeviceConnectionStatus())));
             }
         }
-        if (predicates.getGroupId() != null) {
-            andPred = andPred.and(new AttributePredicate<KapuaId>(DevicePredicates.GROUP_ID, KapuaEid.parseCompactId(predicates.getGroupId())));
+        if (predicates.getGroupDevice() != null) {
+            switch (predicates.getGroupDeviceStatusEnum()) {
```
Comment (muros-ct) [inline console/module/device/src/main/java/org/eclipse/kapua/app/console/module/device/shared/util/GwtKapuaDeviceModelConverter.java]: same comment, why Status

File: console/module/device/src/main/java/org/eclipse/kapua/app/console/module/device/shared/util/GwtKapuaDeviceModelConverter.java [hunk 5]
```diff
@@ -187,8 +187,16 @@ public static DeviceQuery convertDeviceQuery(PagingLoadConfig loadConfig, GwtDev
...
+            case NO_GROUP:
+                andPred = andPred.and(new AttributePredicate<KapuaId>(DevicePredicates.GROUP_ID, null, Operator.IS_NULL));
+                break;
+            default:
+                if (predicates.getGroupId() != null) {
+                andPred = andPred.and(new AttributePredicate<KapuaId>(DevicePredicates.GROUP_ID, KapuaEid.parseCompactId(predicates.getGroupId())));
```
Comment (muros-ct) [inline console/module/device/src/main/java/org/eclipse/kapua/app/console/module/device/shared/util/GwtKapuaDeviceModelConverter.java]: Code formatting

File: console/module/device/src/main/java/org/eclipse/kapua/app/console/module/device/shared/util/KapuaGwtDeviceModelConverter.java [hunk 6]
```diff
@@ -53,7 +54,11 @@ public static GwtDevice convertDevice(Device device)
         gwtDevice.setDisplayName(device.getDisplayName());
         gwtDevice.setModelId(device.getModelId());
         gwtDevice.setSerialNumber(device.getSerialNumber());
-        gwtDevice.setGroupId(KapuaGwtCommonsModelConverter.convertKapuaId(device.getGroupId()));
+        if (device.getGroupId() != null) {
+            gwtDevice.setGroupDevice(GwtGroupDevice.NO_GROUP.name());
+        } else {
```
Comment (muros-ct) [inline console/module/device/src/main/java/org/eclipse/kapua/app/console/module/device/shared/util/KapuaGwtDeviceModelConverter.java]: code formatting

**Comment by LeoNerdoG** [comment]
I have verified that these changes do not affect functionality, so it is still OK to merge, if changes are OK.

**Pull request merged**